en/2-wallet/README.md

```diff
@@ -14,3 +14,351 @@
+
+This encrypted key provides an additional layer of security to the raw private so that the attacker would require would both the encrypted key and the password in order to access that funds. This is good, but it is often that one might need to have multiple wallets, which means that they have multiple keys. Storing each NEP2 encrypted key would be very cumbersome, so instead we can create a file structure that would allow for all of these encrypted keys to be stored in the same place. 
+
+This standards allows for a standard way of importing wallets into various blockchain clients and contains the additional secruity guarantees of the NEP2 format.
+
+A full specification of the file format can be found here. (https://github.com/neo-project/proposals/blob/master/nep-6.mediawiki). It follows a JSON structure that contains information about the private/public key pairs as well as metadata about each account. The metadata details contains info like which wallet should be used as default, the encryption parameters, and any other relevant metadata. 
+
+The NEP6 file also supports watch only addresses. Watch only addresses do not contain any information related to the private key, which maybe useful if the account is stored seperately in a more secure location.
+
+## Contract Account
+NEO also supports more sophisticated Account Types. In these cases the funds are not associated with a single user but stored in a smart contract. The contract would create special rules on what is required in order for funds to to be withdrawn from this account. The most common case of this type of account is a multi signature account. A multi signature account requires that N of X people provides signatures for the transaction in order to transfer funds. For example 2 of 3 of the account owners must sign in order for the funds to be withdrawn. 
+
+We can generate a simple contract for this account using NEO op codes
+
+Suppose we want to create a multisignature contract account for THREE different persons (public keys):
+
+```
+//pubkey1
+036245f426b4522e8a2901be6ccc1f71e37dc376726cc6665d80c5997e240568fb
+
+//pubkey2
+0303897394935bb5418b1c1c4cf35513e276c6bd313ddd1330f113ec3dc34fbd0d
+
+//pubkey3
+02e2baf21e36df2007189d05b9e682f4192a101dcdf07eed7d6313625a930874b4
+```
+
+We want to have at least TWO of them to sign the transactions. So we must create a custom script for this purpose. The script is as follows
+```
+// minimum number of signatures (2)
+PUSH OPCODE 52
+
+// attach all of the pubkeys
+PUSH PUBKEY 1
+PUSH PUBKEY 2
+PUSH PUBKEY 3
+
+//33 bytes
+PUSH OPCODE 21
+
+//total number of people (3)
+PUSH OPCODE 53
+
+//CHECK MULTISIHG
+PUSH OPCODE AE
+```
+
```

Comment:
I think there is a error here.
The code of creating a multisignature as follows:
        public static byte[] CreateMultiSigRedeemScript(int m, params ECPoint[] publicKeys)
        {
            if (!(1 <= m && m <= publicKeys.Length && publicKeys.Length <= 1024))
                throw new ArgumentException();
            using (ScriptBuilder sb = new ScriptBuilder())
            {
                sb.EmitPush(m);
                foreach (ECPoint publicKey in publicKeys.OrderBy(p => p))
                {
                    sb.EmitPush(publicKey.EncodePoint(true));
                }
                sb.EmitPush(publicKeys.Length);
                sb.Emit(OpCode.CHECKMULTISIG);
                return sb.ToArray();
            }
        }

So "PUSH OPCODE 21" is not  happen in the process